spring-xd-dirt/src/main/java/org/springframework/xd/dirt/module/WriteCapableArchiveModuleRegistry.java

```diff
@@ -42,13 +42,13 @@ public WriteCapableArchiveModuleRegistry(String root) {
 	@Override
 	public boolean delete(ModuleDefinition definition) {
 		try {
-			File module = targetModuleLocation(definition);
-			boolean wasFile = module.exists() && !module.isDirectory() && module.delete();
-			if (wasFile) {
-				return true;
-			} else {
-				String filename = module.getName();
-				File asDir = new File(module.getParentFile(), filename.substring(0, filename.lastIndexOf('.')));
+			File moduleJarFile = targetModuleLocation(definition);
+			if (moduleJarFile.exists()) {
```

Comment:
`moduleJarFile.exists()` would return true for a directory, and then fail in the if() body. The if condition should have `!moduleJarFile.isDirectory()` in addition I believe